ement-room.el

```diff
@@ -2847,35 +2853,35 @@ MESSAGE is the message STRING was contained in."
         :then (lambda (data)
                 (when (buffer-live-p saved-buffer)
                   (with-current-buffer saved-buffer
-                    (ement-room--insert-preview message
+                    (ement-room--insert-preview event
                                                 related-event
-                                                num
+                                                link
                                                 (progn (unless (alist-get 'og:url data)
                                                          (push (cons 'og:url link) data))
                                                        (unless (alist-get 'og:title data)
                                                          (push (cons 'og:title "[untitled page]") data))
                                                        data)))))
         :else (lambda (why) (ement-debug "Error in fetching preview: " why))))))
 
-(defun ement-room--insert-preview (message related num data)
-  "Insert link preview for DATA into MESSAGE from node RELATED if it isn't already present.
-NUM is the index for this link."
+(defun ement-room--insert-preview (message related key data)
+  "Insert link preview for HTTP response DATA into MESSAGE from node RELATED,
```

Comment:
Note that docstrings' first lines should be a single sentence.  You can use `M-x checkdoc` to help with these guidelines.